pkg/streamingpromql/operators/functions/range_vectors.go

```diff
@@ -533,3 +533,61 @@ func linearRegression(head, tail []promql.FPoint, interceptTime int64) (slope, i
 	intercept = sumY/n - slope*sumX/n
 	return slope, intercept
 }
+
+var Irate = FunctionOverRangeVectorDefinition{
+	SeriesMetadataFunction: DropSeriesName,
+	StepFunc:               irate(true),
+}
+
+var Idelta = FunctionOverRangeVectorDefinition{
+	SeriesMetadataFunction: DropSeriesName,
+	StepFunc:               irate(false),
+}
+
+func irate(isRate bool) RangeVectorStepFunction {
```

Comment:
(nit)
```suggestion
func irateIdelta(isRate bool) RangeVectorStepFunction {
```

Reply:
Addressed in 5383c9cc0dcf10fd046b00314d929a42f753b263